[SPARK-35591][PYTHON][DOCS] Rename "Koalas" to "pandas API on Spark" in the documents (#32835)
apache/spark: python/docs/source/development/ps_contributing.rst

Running Tests
=============

> There is a script `./dev/pytest` which is exactly same as `pytest` but with some default settings to run pandas APIs on Spark tests easily.

Reviewer: ```suggestion
There is a script `./dev/pytest` which is exactly same as `pytest` but with some default settings to run the tests easily.
```

Doctest Conventions
===================

When writing doctests, usually the doctests in pandas are converted into pandas APIs on Spark to make sure the same codes work in pandas APIs on Spark.
In general, doctests should be grouped logically by separating a newline.

For instance, the first block is for the statements for preparation, the second block is for using the function with a specific argument,
and third block is for another argument. As a example, please refer `DataFrame.rsub <https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.rsub.html#pandas.DataFrame.rsub>`_ in pandas.

> These blocks should be consistently separated in pandas APIs on Spark, and more doctests should be added if the coverage of the doctests or the number of examples to show is not enough even though they are different from pandas'.

Reviewer: ```suggestion
These blocks should be consistently separated in pandas-on-Spark doctests, and more doctests should be added if the coverage of the doctests or the number of examples to show is not enough even though they are different from pandas'.
```

Release Guide
=============

Release Cadence
---------------

> Pandas APIs on Spark 1.8.0 is the last minor release because pandas APIs on Spark will be officially included to PySpark.

Reviewer: ```suggestion
Koalas 1.8.0 is the last minor release because Koalas will be officially included to PySpark.
```
